x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -142,9 +160,24 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 				return fmt.Errorf("fetch variable mode failed: %w", err)
 			}
 
-			m.reportEvent(ms, reporter, q.Query)
+			if m.Config.MergeQueries {
+				for k, v := range ms {
+					_, ok := merged[k]
+					if ok {
+						m.Logger().Warn("overwriting duplicate metrics: ", k)
+					}
+					merged[k] = v
+				}
+			} else {
+				// Report immediately for non-merged cases.
+				m.reportEvent(ms, reporter, q.Query)
+			}
 		}
 	}
+	if m.Config.MergeQueries {
+		// Report here for merged case.
+		m.reportEvent(merged, reporter, "MULTIPLE")
```

Comment:
I think I'd prefer not to set `sql.query` on this case. Maybe you can pass an empty value here, and make `reportEvent` to don't set this value in this case.
```suggestion
		m.reportEvent(merged, reporter, "")
```

Reply:
Made it empty.